### PR TITLE
Propel Bridge

### PR DESCRIPTION
**Pull request by liamsorsby**

### CODE REVIEW

**Comment by liamsorsby** [comment]
@makasim made the pull request if you can have a look and let me know of any changes that are needed

**Comment by liamsorsby** [comment]
Seems to be failing on including the Klarna factory into the project. Is this a known issue at the moment? Or is this something to do with my pull request?

**Comment by makasim** [comment]
nope, it was broken in master, I address it asap.

**Comment by makasim** [comment]
Why did you moved commands: https://www.dropbox.com/s/b75489zurk4vzwt/Screenshot%202015-06-09%2011.00.28.png?dl=0

File: DependencyInjection/Factory/Storage/Propel1StorageFactory.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        return "propel1";
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public function addConfiguration(ArrayNodeDefinition $builder)
+    {
+        parent::addConfiguration($builder);     
+    }
```
Comment (makasim) [inline DependencyInjection/Factory/Storage/Propel1StorageFactory.php]: you can remove the method definition at all.

File: Resources/config/storage/propel1.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <parameters>
+        <parameter key="payum.storage.propel1.class">Payum\Core\Bridge\Propel\Storage\Propel1Storage</parameter>
+    </parameters>
+
+    <services>
+        <service id="payum.storage.propel1" class="%payum.storage.propel1.class%" public="false" abstract="true">
```
Comment (makasim) [inline Resources/config/storage/propel1.xml]: set the class here, do not use parameter.

File: Tests/DependencyInjection/Factory/Storage/Propel1StorageFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        $tb = new TreeBuilder();
+        $rootNode = $tb->root('foo');
+
+        $factory->addConfiguration($rootNode);
+
+        $processor = new Processor();
+        $config = $processor->process($tb->buildTree(), array(array(
+            'id_property' => 'id',
+            'storage_dir' => '/the/path/to/store/models',
```
Comment (makasim) [inline Tests/DependencyInjection/Factory/Storage/Propel1StorageFactoryTest.php]: should be here

**Comment by liamsorsby** [comment]
@makasim I have no idea why these have been moved, I run a mv command which must have been wrong. I'll edit these now, I've also added the docs so I'll update the pull request.

**Comment by liamsorsby** [comment]
@makasim I've changed the files now.

**Comment by liamsorsby** [comment]
@makasim I've rebased the code and made a few changes. Can you have a look and let me know if any further changes are needed. The tests for the Propel2StorageFactoryTest and Propel1StorageFactoryTest looks a little minimal and may need a few more things to test but I'm unsure what needs testing. If you could let me know that would be great.

File: Resources/config/storage/propel2.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="payum.storage.propel2" class="Payum\Core\Bridge\Propel2\Storage\Propel2Storage" public="false" abstract="true">
+        <argument><!-- should be set in Propel1StorageFactory --></argument>
```
Comment (makasim) [inline Resources/config/storage/propel2.xml]: Propel1StorageFactory -> Propel2StorageFactory

File: Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+use Payum\Bundle\PayumBundle\DependencyInjection\Factory\Storage\Propel2StorageFactory;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\Processor;
+
+class Propel2StorageFactoryTest extends \PHPUnit_Framework_TestCase {
+
+    /**
+     * @test
+     */
+    public function shouldBeSubClassOfAbstractStorageFactory() {
```
Comment (makasim) [inline Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php]: `{` at the new line

File: Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+/* Copyright Astral Game Servers Ltd 2013-2014
+ * Developed By
+ * Liam Sorsby
+ */
```
Comment (makasim) [inline Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php]: remove it.

**Comment by liamsorsby** [comment]
@makasim sorry didn't see the licence headers that had been put on. These have been changed

File: Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Payum\Bundle\PayumBundle\Tests\DependencyInjection\Factory\Storage;
+
+use Payum\Bundle\PayumBundle\DependencyInjection\Factory\Storage\Propel2StorageFactory;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\Processor;
+
+class Propel2StorageFactoryTest extends \PHPUnit_Framework_TestCase {
```
Comment (makasim) [inline Tests/DependencyInjection/Factory/Storage/Propel2StorageFactoryTest.php]: { - here

File: DependencyInjection/Factory/Storage/Propel2StorageFactory.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+namespace Payum\Bundle\PayumBundle\DependencyInjection\Factory\Storage;
+
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+use Symfony\Component\Config\FileLocator;
+
+
+class Propel2StorageFactory  extends AbstractStorageFactory{
```
Comment (makasim) [inline DependencyInjection/Factory/Storage/Propel2StorageFactory.php]: { - here, and we are good

File: DependencyInjection/Factory/Storage/Propel1StorageFactory.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+namespace Payum\Bundle\PayumBundle\DependencyInjection\Factory\Storage;
+
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+use Symfony\Component\Config\FileLocator;
+
+
+class Propel1StorageFactory  extends AbstractStorageFactory{
```
Comment (makasim) [inline DependencyInjection/Factory/Storage/Propel1StorageFactory.php]: ah, here too

**Comment by liamsorsby** [comment]
@makasim There we go, that should do it.

**Comment by liamsorsby** [comment]
@makasim are the changes to the doc's sufficient aswel?

**Comment by makasim** [comment]
@liamsorsby  Thanks!

**Pull request merged**